Makefile

```diff
@@ -1,18 +1,36 @@
-BUILD_NUMBER := 1
-CP_VERSION := 3.3.0-SNAPSHOT
+# You can override vars like REPOSITORY in a local.make file
+-include local.make
+
+# Bump this on subsequent build, reset on new version or public release. Inherit from env for CI builds.
+BUILD_NUMBER ?= 1
+
+CONFLUENT_MAJOR_VERSION ?= 3
+CONFLUENT_MINOR_VERSION ?= 3
+CONFLUENT_PATCH_VERSION ?= 0
+
+CONFLUENT_VERSION ?= ${CONFLUENT_MAJOR_VERSION}.${CONFLUENT_MINOR_VERSION}.${CONFLUENT_PATCH_VERSION}
```

Comment:
I think my confusion is from where we don't include it in the version that isn't a case where we need only a subset of the version (e.g. when we only need the `${CONFLUENT_MAJOR_VERSION}.${CONFLUENT_MINOR_VERSION}` and when those labels would differ between all those components. Is there sometime other than `-SNAPSHOT` where these labels are something different?